File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    answerdict = {}
    for i in range(len(keymap)) :
        for j in range(len(keymap[i])) :
            if keymap[i][j] not in answerdict:
                answerdict[keymap[i][j]] = []
            
            answerdict[keymap[i][j]].append(j+1)
    
    
    for i in range(len(targets)) :
        value = 0
        for j in range(len(targets[i])) :
            if targets[i][j] not in answerdict :
                value = -1
                break
            else:
                value+= min(answerdict[targets[i][j]])
        answer.append(value)
    
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int [targets.length];
        
        Map <Character,Integer> map = new HashMap<>();
        
        for (int i = 0 ; i<keymap.length; i++)
        {
            for (int j = 0 ; j< keymap[i].length();j++)
            {
                if (!map.keySet().contains(keymap[i].charAt(j)))
                {
                    map.put(keymap[i].charAt(j),j);
                } else {
                    int min = Math.min(j,map.get(keymap[i].charAt(j)));
                    map.put(keymap[i].charAt(j),min);
                }
            }
        }
      
            
        for (int i = 0 ;i<targets.length; i++)
        {
              int ans = 0 ;
            int before = -1;
            for (int j = 0 ; j<targets[i].length();j++)
            {  
                char now = targets[i].charAt(j);
                
                if (map.keySet().contains(now))
                {
                    ans += (map.get(now)+1);
                } else {
                    ans = 0;
                    break;
                }
                
            }
            
                         
        ans = ans == 0? -1 : ans;
        answer[i] = ans; 
        }

        
        
        return answer;
    }
}